server/sonar-server/src/main/java/org/sonar/server/permission/GroupPermissionChanger.java

```diff
@@ -74,9 +78,16 @@ private boolean removePermission(DbSession dbSession, GroupPermissionChange chan
       change.getOrganizationUuid(),
       change.getGroupIdOrAnyone().getId(),
       change.getNullableProjectId());
+    updateRootFlag(dbSession, change);
     return true;
   }
 
+  private void updateRootFlag(DbSession dbSession, GroupPermissionChange change) {
+    if (SYSTEM_ADMIN.equals(change.getPermission()) && !change.getGroupIdOrAnyone().isAnyone()) {
```

Comment:
the root flag relates to global permissions only. You can ignore modifications of project permissions by adding the condition on `!change.getProjectId().isPresent()`.